osu.Game/Rulesets/Objects/HitWindows.cs

```diff
@@ -53,14 +51,25 @@ public class HitWindows
         public double Miss { get; protected set; }
 
         /// <summary>
-        /// Whether it's possible to achieve a <see cref="HitResult.Perfect"/> result.
+        /// Hit window for a non-<see cref="HitResult.Miss"/> result.
         /// </summary>
-        public bool AllowsPerfect;
+        protected virtual double SuccessfulHitWindow => Meh;
```

Comment:
How about this returning a `HitResult` type, and doing `timeOffset <= HalfWindowFor(SuccessfulHitWindow)`?

Reply:
Would also rename to `SuccessfulHitResult`.